### PR TITLE
detect/mark:Fix unittest into FAIL/PASS API

### PR DESCRIPTION
**Pull request by JananiRamjee**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4052

Describe changes:
- Removed initial test methods
- Used FAIL or PASS macros described in util-unittest.h
- FAIL/PASS macros determine conditionals for return value(0 or 1 respectively)

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by JananiRamjee** [comment]
Hello @inashivb, 
Requesting a feedback for the request. Thank you!

File: src/detect-mark.c [hunk 1]
```diff
@@ -253,81 +253,77 @@ static int DetectMarkPacket(DetectEngineThreadCtx *det_ctx, Packet *p,
...
-    return 1;
+    PASS;
 }
 
 /**
  * \test MarkTestParse04 is a test for a invalid mark value
  *
- *  \retval 1 on succces
- *  \retval 0 on failure
+ * \FAIL/PASS API
```
Comment (victorjulien) [inline src/detect-mark.c:313]: these values are self explaining and don't need per function documentation

**Review by victorjulien — CHANGES_REQUESTED** [review]
The git commit message can be a bit more to the point. No need to explain the value of the PASS/FAIL macros for example. I would suggest something like "detect/mark: convert unittests to use PASS/FAIL API"

File: src/detect-mark.c (hunk 1 above)
Comment (JananiRamjee) [reply src/detect-mark.c:313]: Sure!
Shall I commit the changes here, or give a separate pull request?

Comment (victorjulien) [reply src/detect-mark.c:313]: Our workflow is to create a new branch + new PR for each "cycle" of the review process. So please create a new branch, new PR, then add a reference to the new PR in this old PR and close this one.

**Comment by JananiRamjee** [comment]
Replaced with #5484

**Pull request closed without merging**